File: Instructor Python/Collections/Python-Array-Double/Python/Start.py
from ArrayDouble import ArrayDouble


def start():
    # Get random int
    arr = ArrayDouble.rand(20, -1, 1)
    # Print Array
    ArrayDouble.print(arr)
    # Sort Array
    print("Sort Array -------------------------")
    arr.sort()
    ArrayDouble.print(arr)
    print("filter > 0-------------------------")
    arr3 = filter(lambda c: c > 0, arr)
    ArrayDouble.print(arr3)
    print("filter <=0-------------------------")
    arr4 = filter(lambda c: c <= 0, arr)
    ArrayDouble.print(arr4)
    print("Sum -------------------------")
    sumValue = sum(arr)
    print(f"Sum = {sumValue:f}")


if __name__ == "__main__":
    start()

File: Instructor Java/collections/Java-Array-Double/Java/Start.java
import java.util.Arrays;

/**
 * Start
 */
public class Start {

  public static void main(String[] args) {
    // Create Array
    var arr = ArrayDouble.rand(20, -1, 1);
    // print Array
    ArrayDouble.print(arr);
    // Sort Array
    System.out.println("Sort Array -------------------------");
    var arr2 = Arrays.stream(arr).sorted().toArray();
    ArrayDouble.print(arr2);
    System.out.println("filter > 0-------------------------");
    var arr3 = Arrays.stream(arr).filter(c -> c > 0).toArray();
    ArrayDouble.print(arr3);
    System.out.println("filter <= 0-------------------------");
    var arr4 = Arrays.stream(arr).filter(c -> c <= 0).toArray();
    ArrayDouble.print(arr4);
    System.out.println("Sum -------------");
    var sum = Arrays.stream(arr).sum();
    System.out.printf("sum =%f\n ", sum);
    System.out.println("Sum using reduce-------------");
    var reduce = Arrays.stream(arr).reduce(0.0, (element, total) -> total + element);
    System.out.printf("reduce =%f\n ", reduce);
  }
}
